packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -209,25 +236,36 @@ abstract class ResidentRunner {
     _loggingSubscription = null;
   }
 
-  Future<Null> connectToServiceProtocol(Uri uri, {String isolateFilter}) async {
-    if (!debuggingOptions.debuggingEnabled)
+  Future<Null> connectToServiceProtocol(List<Uri> uris,
+      {String isolateFilter}) async {
+    if (!debuggingOptions.debuggingEnabled) {
       return new Future<Null>.error('Error the service protocol is not enabled.');
-    vmService = VMService.connect(uri);
-    printTrace('Connected to service protocol: $uri');
-    await vmService.getVM();
+    }
+    final List<VMService> services = new List<VMService>(uris.length);
+    for (int i = 0; i < uris.length; i++) {
+      services[i] = VMService.connect(uris[i]);
+      printTrace('Connected to service protocol: ${uris[i]}');
+    }
+    vmServices = services;
+    for (VMService service in services) {
+      await service.getVM();
+    }
+
+    this.isolateFilter = isolateFilter;
 
     // Refresh the view list, and wait a bit for the list to populate.
-    await vmService.waitForViews();
-    currentView = (isolateFilter == null)
-                ? vmService.vm.firstView
-                : vmService.vm.firstViewWithName(isolateFilter);
+    for (VMService service in services) {
+      await service.waitForViews();
+    }
     if (currentView == null)
       throwToolExit('No Flutter view is available');
 
     // Listen for service protocol connection to close.
-    vmService.done.then<Null>(
-        _serviceProtocolDone,
-        onError: _serviceProtocolError).whenComplete(appFinished);
+    for (VMService service in services) {
+      service.done.then<Null>(
```

Comment:
nit: if you have a newline after the ( please put a newline before the matching )